### PR TITLE
feat(editor): updated n8n-menu component

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
Closes N8N-4736

### CODE REVIEW

File: packages/design-system/src/components/N8nMenu/Menu.vue [hunk 1]
```diff
@@ -1,22 +1,83 @@
...
+						:compact="collapsed"
+						:popperClass="$style.submenuPopper"
+						:tooltipDelay="tooltipDelay"
+						:mode="mode"
+						:activeTab="activeTab"
+						@click="onSelect"
+					/>
+				</el-menu>
+			</div>
+			<div :class="{[$style.lowerContent]: true, ['pb-xs']: $slots.menuSuffix === undefined}">
```
Comment (cstuncsik) [inline packages/design-system/src/components/N8nMenu/Menu.vue:33]: just a nitpick (on code style)
```suggestion
			<div :class="{[$style.lowerContent]: true, 'pb-xs': !$slots.menuSuffix}">
```

File: packages/design-system/src/components/N8nMenu/Menu.vue [hunk 2]
```diff
@@ -26,55 +87,96 @@ export default Vue.extend({
...
+		items: {
+			type: Array as PropType<IMenuItem[]>,
 		},
 	},
-	components: {
-		ElMenu, // eslint-disable-line @typescript-eslint/no-unsafe-assignment
+	mounted() {
+		if (this.mode === 'router') {
+			const found = this.items.find(item => {
+				return item.activateOnRouteNames !== undefined && (item.activateOnRouteNames ).includes(this.$route.name || '') ||
```
Comment (cstuncsik) [inline packages/design-system/src/components/N8nMenu/Menu.vue:110]: again minor thing
```suggestion
				return Array.isArray(item.activateOnRouteNames) && item.activateOnRouteNames.includes(this.$route.name || '') ||
```

File: packages/design-system/src/components/N8nMenu/Menu.vue [hunk 3]
```diff
@@ -26,55 +87,96 @@ export default Vue.extend({
...
+					item.activateOnRoutePaths !== undefined && (item.activateOnRoutePaths).includes(this.$route.path);
+			});
+			this.activeTab = found ? found.id : '';
+		} else {
+			this.activeTab =  this.items.length > 0 ? this.items[0].id : '';
+		}
+	},
+	computed: {
+		upperMenuItems(): IMenuItem[] {
+			return this.items.filter((item: IMenuItem) => item.position === 'top' && item.available !== false);
```
Comment (cstuncsik) [inline packages/design-system/src/components/N8nMenu/Menu.vue:120]: guessing about the name `item.available` is a `boolean` so
```suggestion
			return this.items.filter((item: IMenuItem) => item.position === 'top' && item.available);
```

File: packages/design-system/src/components/N8nMenuItem/MenuItem.vue [hunk 4]
```diff
@@ -1,7 +1,268 @@
...
+			default: 'router',
+			validator: (value: string): boolean => ['router', 'tabs'].includes(value),
+		},
+		activeTab: {
+			type: String,
+		},
+	},
+	computed: {
+		availableChildren(): IMenuItem[] {
+			return this.item.children ? this.item.children.filter(child => child.available !== false) : [];
```
Comment (cstuncsik) [inline packages/design-system/src/components/N8nMenuItem/MenuItem.vue:99]: same here for `boolean`
```suggestion
			return this.item.children ? this.item.children.filter(child => child.available) : [];
```

File: packages/design-system/src/components/N8nMenuItem/MenuItem.vue [hunk 5]
```diff
@@ -1,7 +1,268 @@
...
+		},
+	},
+	methods: {
+		isItemActive(item: IMenuItem): boolean {
+			const isItemActive = this.isActive(item);
+			const childActive = item.children !== undefined && item.children.find(child => this.isActive(child)) !== undefined;
```
Comment (cstuncsik) [inline packages/design-system/src/components/N8nMenuItem/MenuItem.vue:105]: I would make every expression evaluate to `boolean` and `Array.find` returns the found element or undefined so maybe better if you check if there is any active item (also variable name could be `hasActiveChild`)
```suggestion
			const hasActiveChild = Array.isArray(item.children) && item.children.some(child => this.isActive(child));
```

File: packages/design-system/src/components/N8nMenuItem/MenuItem.vue [hunk 6]
```diff
@@ -1,7 +1,268 @@
...
+			return isItemActive || childActive;
+		},
+		isActive(item: IMenuItem): boolean {
+			if (this.mode === 'router') {
+				if (item.activateOnRoutePaths) {
+					return item.activateOnRoutePaths !== undefined && item.activateOnRoutePaths.includes(this.$route.path);
```
Comment (cstuncsik) [inline packages/design-system/src/components/N8nMenuItem/MenuItem.vue:111]:
```suggestion
					return Array.isArray(item.activateOnRoutePaths) && item.activateOnRoutePaths.includes(this.$route.path);
```

File: packages/design-system/src/components/N8nMenu/Menu.vue [hunk 7]
```diff
@@ -26,55 +87,96 @@ export default Vue.extend({
...
+		},
+		lowerMenuItems(): IMenuItem[] {
+			return this.items.filter((item: IMenuItem) => item.position === 'bottom' && item.available !== false);
```
Comment (cstuncsik) [inline packages/design-system/src/components/N8nMenu/Menu.vue:123]: `item.available !== false` should be just `item.available`

**Review by cstuncsik — DISMISSED** [review]

**Review by cstuncsik — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.198.0`

**Pull request merged**